doc/tools/docgram/doc_grammar.ml

```diff
@@ -0,0 +1,1487 @@
+      end
+  in
+  List.rev (nt_closure_r [] [start])
+
+let header = "--------------------------------------------"
+let nt_subset_in_orig_order g nts =
+  let subset = StringSet.of_list nts in
+  List.filter (fun nt -> StringSet.mem nt subset) !g.order
+
+let print_chunk out g seen fmt title starts ends =
+  fprintf out "\n\n%s:\n%s\n" title header;
+  List.iter (fun start ->
+      let nts = (nt_closure g start ends) in
+      print_in_order out g fmt (nt_subset_in_orig_order g nts) !seen;
+      seen := StringSet.union !seen (StringSet.of_list nts))
+    starts
+
+let print_chunks g out fmt () =
+  let seen = ref StringSet.empty in
+  print_chunk out g seen fmt "lconstr" ["lconstr"] ["binder_constr"; "tactic_expr5"];
+  print_chunk out g seen fmt "Gallina syntax of terms" ["binder_constr"] ["tactic_expr5"];
+  print_chunk out g seen fmt "Gallina The Vernacular" ["gallina"] ["tactic_expr5"];
+  print_chunk out g seen fmt "intropattern_list_opt" ["intropattern_list"; "or_and_intropattern_loc"] ["operconstr"; "tactic_expr5"];
+  print_chunk out g seen fmt "simple_tactic" ["simple_tactic"]
+    ["tactic_expr5"; "tactic_expr3"; "tactic_expr2"; "tactic_expr1"; "tactic_expr0"];
+
+  (*print_chunk out g seen fmt "Ltac" ["tactic_expr5"] [];*)
+  print_chunk out g seen fmt "Ltac" ["tactic_expr5"] ["tactic_expr4"];
+  print_chunk out g seen fmt "Ltac 4" ["tactic_expr4"] ["tactic_expr3"; "tactic_expr2"];
+  print_chunk out g seen fmt "Ltac 3" ["tactic_expr3"] ["tactic_expr2"];
+  print_chunk out g seen fmt "Ltac 2" ["tactic_expr2"] ["tactic_expr1"];
+  print_chunk out g seen fmt "Ltac 1" ["tactic_expr1"] ["tactic_expr0"];
+  print_chunk out g seen fmt "Ltac 0" ["tactic_expr0"] [];
+
+
+  print_chunk out g seen fmt "command" ["command"] [];
+  print_chunk out g seen fmt "vernac_toplevel" ["vernac_toplevel"] [];
+  print_chunk out g seen fmt "vernac_control" ["vernac_control"] []
+
+  (*
+    let ssr_tops = ["ssr_dthen"; "ssr_else"; "ssr_mpat"; "ssr_rtype"] in
+    seen := StringSet.union !seen (StringSet.of_list ssr_tops);
+
+    print_chunk out g seen fmt "ssrindex" ["ssrindex"] [];
+    print_chunk out g seen fmt "command" ["command"] [];
+    print_chunk out g seen fmt "binder_constr" ["binder_constr"] [];
+    (*print_chunk out g seen fmt "closed_binder" ["closed_binder"] [];*)
+    print_chunk out g seen fmt "gallina_ext" ["gallina_ext"] [];
+    (*print_chunk out g seen fmt "hloc" ["hloc"] [];*)
+    (*print_chunk out g seen fmt "hypident" ["hypident"] [];*)
+    print_chunk out g seen fmt "simple_tactic" ["simple_tactic"] [];
+    print_chunk out g seen fmt "tactic_expr" ["tactic_expr4"; "tactic_expr1"; "tactic_expr0"] [];
+    fprintf out "\n\nRemainder:\n";
+    print_in_order g (List.filter (fun x -> not (StringSet.mem x !seen)) !g.order) StringSet.empty;
+  *)
+
+
+    (*seen := StringSet.diff !seen (StringSet.of_list ssr_tops);*)
+    (*print_chunk out g seen fmt "vernac_toplevel" ["vernac_toplevel"] [];*)
```

Comment:
Left-over commented out code.

Reply:
This was an experiment trying to compute a reasonable order for SSR nonterminals that I set aside for a while.  I thought I might come back to it when we do that chapter.  And once complete, drop the comments or complete the code.  Making local code modifications to do such computations is a useful approach.

Reply:
As long as you remember to come back to this to fix it or remove it, that's fine, but the problem with commented code is that it tends to rot quickly.